keps/2724-topology-aware-scheduling/README.md

```diff
@@ -663,11 +685,23 @@ For a given PodSet Kueue:
   level. Kueue starts the search from the specified level, but if the PodSet
   does not fit, then it tries higher levels in the hierarchy.
 
-Kueue packs pods on domains starting from the domains with the most free capacity. However, Kueue can operate in two modes when it comes to choosing the last domain if there is more than one capable of accommodating the remaining pods:
-- `MostAllocated` - Kueue chooses the domain with the least available resource that is capable of accommodating all the pods to mitigate resource fragmentation
-- `LeastAllocated` - Kueue chooses the domain that has the most available resources, providing better nodes utilization
+Kueue places pods on domains with different algorithms, depending on the annotation and chosen profile:
+- `MostFreeCapacity` algorithm - Kueue selects as many domains as needed (if it meets user's requirement) starting from the one with the most free capacity;
+- `LeastFreeCapacity` algorithm - Kueue selects as many domains as needed (if it meets user's requirement) starting from the one with the least free capacity;
+- `BestFit` algorithm - Kueue selects as many domains as needed (if it meets user's requirement) starting from the one with the most free capacity, but tries to optimize the last domain to leave as few free resources as possible.
+
+Selection of the algorithm depends on TAS profiles expressed by feature gates, and PodSet's annotation:
+
+| featuregate/annotation                   | preferred         | required          | unconstrained     |
+| ---------------------------------------- | ----------------- | ----------------- | ----------------- |
+| None                                     | BestFit           | BestFit           | BestFit           |
+| TASProfileMostFreeCapacity (deprecated)  | MostFreeCapacity  | MostFreeCapacity  | MostFreeCapacity  |
+| TASProfileMixed (deprecated)             | BestFit           | BestFit           | LeastFreeCapacity |
+| TASProfileLeastFreeCapacity (deprecated) | LeastFreeCapacity | LeastFreeCapacity | LeastFreeCapacity |
```

Comment:
So, eventually, do we want to supply only BestFit, right?

Reply:
@PBundyra may want to add something, but my understanding is summarized in https://github.com/kubernetes-sigs/kueue/pull/4542#discussion_r1991039901. Basically BestFit is the only algorithm which works sensibly as a default. However, other algorithms may find some niche applicability, so we keep the feature gates for experimentation. We don't plan to graduate the feature gates, but remove. We either remove the feature gates along with the algorithms, or introduce a dedicated configuration for them (TBD).

Reply:
Uhm, I see. That makes sense. I just want to leave a note in the KEP that we currently expect the BestFit works well in most situations, but we might make the possibility for other deprecated scheduling based on user feedback.

Reply:
+1 to what mimowo said. I'll add clarification